kernel/dynamic.c

```diff
@@ -15,6 +15,16 @@
 
 LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
 
+#if CONFIG_DYNAMIC_THREAD_STACK_SIZE > 0
+#ifdef CONFIG_USERSPACE
+#define DYNAMIC_STACK_SIZE Z_THREAD_STACK_SIZE_ADJUST(CONFIG_DYNAMIC_THREAD_STACK_SIZE)
+#else
+#define DYNAMIC_STACK_SIZE Z_KERNEL_STACK_SIZE_ADJUST(CONFIG_DYNAMIC_THREAD_STACK_SIZE)
+#endif
+#else
+#define DYNAMIC_STACK_SIZE 0
+#endif
+
```

Comment:
Don't do this here please, we have to do it inside the dynamic  when allocating from the heap (because it can receive any size). If you do this, we will end up wasting memory.